plugins/out_azure_kusto/azure_kusto.c

```diff
@@ -220,19 +431,639 @@ flb_sds_t execute_ingest_csl_command(struct flb_azure_kusto *ctx, const char *cs
+{
+    struct azure_kusto_file *chunk;
+    struct mk_list *tmp;
+    struct mk_list *head;
+    struct mk_list *f_head;
+    struct flb_fstore_file *fsf;
+    struct flb_fstore_stream *fs_stream;
+    flb_sds_t payload = NULL;
+    void *final_payload = NULL;
+    size_t final_payload_size = 0;
+    char *buffer = NULL;
+    size_t buffer_size;
+    int ret;
+    int is_compressed = FLB_FALSE;
+    flb_sds_t tag_sds;
+
+    mk_list_foreach(head, &ctx->fs->streams) {
+        fs_stream = mk_list_entry(head, struct flb_fstore_stream, _head);
+        if (fs_stream == ctx->stream_upload) {
+            continue;
+        }
+
+        mk_list_foreach_safe(f_head, tmp, &fs_stream->files) {
+            fsf = mk_list_entry(f_head, struct flb_fstore_file, _head);
+            chunk = fsf->data;
+
+            /* Locked chunks are being processed, skip */
+            if (chunk->locked == FLB_TRUE) {
+                continue;
+            }
+
+            if (chunk->failures >= ctx->scheduler_max_retries) {
+                flb_plg_warn(ctx->ins,
+                             "ingest_all_old_buffer_files :: Chunk for tag %s failed to send %i times, "
+                             "will not retry",
+                             (char *) fsf->meta_buf, ctx->scheduler_max_retries);
+                if (ctx->delete_on_max_upload_error){
+                    azure_kusto_store_file_delete(ctx, chunk);
+                }else{
+                    azure_kusto_store_file_inactive(ctx, chunk);
+                }
+                continue;
+            }
+
+            ret = construct_request_buffer(ctx, NULL, chunk,
+                                           &buffer, &buffer_size);
+            if (ret < 0) {
+                flb_plg_error(ctx->ins,
+                              "ingest_all_old_buffer_files :: Could not construct request buffer for %s",
+                              chunk->file_path);
+                return -1;
+            }
+
+            payload = flb_sds_create_len(buffer, buffer_size);
+            tag_sds = flb_sds_create(fsf->meta_buf);
+            flb_free(buffer);
+
+            /* Compress the JSON payload */
+            if (ctx->compression_enabled == FLB_TRUE) {
+                ret = flb_gzip_compress((void *) payload, flb_sds_len(payload),
+                                        &final_payload, &final_payload_size);
+                if (ret != 0) {
+                    flb_plg_error(ctx->ins,
+                                  "ingest_all_old_buffer_files :: cannot gzip payload");
+                    flb_sds_destroy(payload);
+                    flb_sds_destroy(tag_sds);
+                    return -1;
+                }
+                else {
+                    is_compressed = FLB_TRUE;
+                    flb_plg_debug(ctx->ins, "ingest_all_old_buffer_files :: enabled payload gzip compression");
+                }
+            } else {
```

Comment:
https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#braces-usage-on-conditionals-loops-and-functions